bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/GraphDataProvider.java

```diff
@@ -681,18 +691,95 @@ public void createEmptyAnnotationSpan(Set<PositionCoordinate> selectedCoordinate
                 + " in the first column of the grid, but found a " + n.getClass().getSimpleName());
       }
     }).collect(Collectors.toList());
-    SSpan span = graph.createSpan(tokens);
-    log.debug("Created new span {}.", span);
-    int columnIndex = selectedCoordinates.iterator().next().getColumnPosition();
-    Column column = getColumns().get(columnIndex);
+    return tokens;
+  }
 
-    for (int rowIndex : selectedRows) {
-      column.setRow(rowIndex, span);
+
+  /**
+   * Splits an {@link SSpan} into spans spanning a single token, retaining annotations.
+   * 
+   * @param span The {@link SSpan} to split
+   * @param coordinates The coordinates for the {@link SSpan} to split
+   */
+  public void splitAnnotationSpan(SSpan span, PositionCoordinate[] coordinates) {
+    PositionCoordinate firstCoord = coordinates[0];
+    int firstCol = firstCoord.getColumnPosition();
+    String annoNamespace = getAnnotationNamespace(firstCol);
+    String annoName = getAnnotationName(firstCol);
+    String annoQName = GridHelper.getAnnotationQName(annoNamespace, annoName);
```

Comment:
If the custom helper function is removed, we could use the SaltUtil helper function here.

```suggestion
    String annoQName = SaltUtil.createQName(annoNamespace, annoName);
```